File: python/TwoSum.py
# https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        ja_vistos = {}

        for i, n in enumerate(nums):
            complement = target - n
            if complement in ja_vistos:
                return [ja_vistos[complement], i]
            else:
                ja_vistos[n] = i 

File: golang/TwoSum.go
func twoSum(nums []int, target int) []int {
    seen := make(map[int]int)

    for i, v := range nums {
        complement := target - v

        if index, ok := seen[complement]; ok {
            return []int{index, i}
        }

        seen[v] = i
    }

    return nil
}